File: ASP/CRMSolution/CRMSolution/Data/Models/Order.cs
namespace CRMSolution.Data.Models;

public class Order
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid ClientId { get; set; }
    public Client Client { get; set; }
    
    public ICollection<Tasks> Tasks { get; set; } = new List<Tasks>();
}

public enum OrderStatus
{
    New,
    Processing,
    Completed
}

File: ASP/CRMFront/src/types/Order.ts
import { Client } from "./Client";
import { Task } from "./Task";
import { User } from "./User";

export interface Order {
    id: number;
    orderId: number,
    totalAmount: string,
    createdAt: Date,
    tasks: Task[],
    orderStatus: OrderStatus,
    user: User[],
    client: Client,
}

export enum OrderStatus
{
    New,
    Processing,
    Completed
}